pages/DSNP/Graph.md

```diff
@@ -2,29 +2,126 @@
 
 This specification describes the social network graph and how it is represented in the protocol.
 
-In this context a [social graph](https://en.wikipedia.org/wiki/Social_graph) means a graph that represents social relations between entities.
+In this context a [social graph](https://en.wikipedia.org/wiki/Social_graph) means a graph that represents social relationships between entities.
 
-The DSNP graph represents nodes as [DSNP User Ids](Identifiers.md#dsnp-user-id).
+## Nodes
 
-## Public Graph
+Users are represented in the DSNP graph by nodes uniquely identified by their [DSNP User Ids](Identifiers.md#dsnp-user-id).
+Relationships between users are the graph edges.
 
-The public graph is anchored in state transitions recorded in [Graph Change Announcements](Types/GraphChange.md).
-Playing all state transitions in [order](Announcements.md#ordering-announcements) will generate the current state of the public graph.
+## Edges
 
-## Non-Normative
+Graph edges that originate from a user are stored as DSNP [User Data](UserData.md) associated with a given user.
 
-### Private Graph
+Each graph edge consists of the following data:
+* DSNP User Id (64-bit unsigned integer)
+  * <mark>To allow for optimal compression, User Ids are stored using the <tt>long</tt> type in Avro schema, which is a 64-bit _signed_ integer.
+    Care should be taken to ensure that User Id values greater than or equal to 2<sup>63</sup>, where used by an implementation, are converted correctly between signed and unsigned representations.</mark>
+* Timestamp when relationship was created, in seconds since the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time)
 
-Implementation coming in the future. See the [DSNP Whitepaper](https://github.com/LibertyDSNP/papers) for foundational ideas.
+GraphEdge objects MUST conform to the following [Avro](https://avro.apache.org) schema:
 
-### Following
+```
+{
+    "namespace": "org.dsnp.userdata",
+    "name": "GraphEdge",
+    "type": "record",
+    "doc": "A relationship to another DSNP user",
+    "fields": [
+        {
+            "name": "userId",
+            "type": "long",
+            "doc": "DSNP User Id of object of relationship"
+        },
+        {
+            "name": "since",
+            "type": "long",
+            "doc": "Timestamp in Unix epoch seconds when this relationship was originally established"
+        }
+    ]
+}
+```
 
-A "follow" is the act of publicly following a [DSNP User Identity](Identifiers.md#dsnp-user-id), which results in adding this DSNP User Id to a user's social graph.
+## User Data Sets
 
```

Comment:
Instead of mentioning data type information here, should that just move over to `UserData.md` and reference a "See [User Data] for details"?